install.sh

```diff
@@ -1001,17 +1001,11 @@ case "${SHELL}" in
     ;;
 esac
 
-# `which` is a shell function defined above.
-# shellcheck disable=SC2230
-if [[ "$(which brew)" != "${HOMEBREW_PREFIX}/bin/brew" ]]
-then
-  cat <<EOS
-- Run these three commands in your terminal to add Homebrew to your ${tty_bold}PATH${tty_reset}:
-    echo '# Set PATH, MANPATH, etc., for Homebrew.' >> ${shell_profile}
-    echo 'eval "\$(${HOMEBREW_PREFIX}/bin/brew shellenv)"' >> ${shell_profile}
+cat <<EOS
+- Run these two commands in your terminal to add Homebrew to your ${tty_bold}PATH${tty_reset}:
+    (echo; echo 'eval "\$(${HOMEBREW_PREFIX}/bin/brew shellenv)"') >> ${shell_profile}
```

Comment:
Shall we consider using `{ }` here (with appropriate changes to the syntax inside `{ }`), which groups commands in the current subshell, instead of `( )`, which starts a new subshell?

Reply:
One thing I'm keeping in mind is what happens when the average user _manually types in the commands incorrectly_, instead of copy-pasting them as they should. (It happens, more frequently than any of us might like.)

Between `( )` and `{ }`, the latter has two significant gotchas: both the space after the opening brace, and the semicolon before the closing one, are _not_ optional. Leaving either one out just results in confusing error messages:
* missing space after open brace: `bash: syntax error near unexpected token '}'`
* missing semicolon before close brace: bare input continuation prompt (`>`)

`( )` is less sensitive to user typos in this respect.

Reply:
> `( )` is less sensitive to user typos in this respect.

👍🏻 to that for me then. The e.g. performance implications are irrelevant in this context.

Reply:
> what happens when the average user _manually types in the commands incorrectly_, instead of copy-pasting them as they should. (It happens, more frequently than any of us might like.)

Out of curiosity: what is this claim based on?